docs/trace/customizing-the-sdk/README.md

```diff
@@ -485,4 +483,191 @@ shutdown.
+
+  ```sh
+  dotnet run --OTEL_SERVICE_NAME "MyService"
+  ```
+
+**Note** The [.NET
+  Configuration](https://learn.microsoft.com/dotnet/core/extensions/configuration)
+  pattern is hierarchical meaning the order of registered configuration sources
+  controls which value will seen by the SDK when it is defined in multiple
+  sources.
+
+### Using the .NET Options pattern to configure the SDK
+
+Options are typically simple classes containing only properties with public
+"getters" and "setters" (aka POCOs) and have "Options" at the end of the class
+name. These options classes are primarily used when interacting with the
+`TracerProviderBuilder` to control settings and features of the different SDK
+components.
+
+Options classes can always be configured through code but users typically want to
+control key settings through configuration.
+
+The following example shows how to configure `JaegerExporterOptions` by binding
+to an `IConfiguration` section.
+
+Json config file (usually appsettings.json):
+
+```json
+{
+  "OpenTelemetry": {
+    "Jaeger": {
+      "Protocol": "UdpCompactThrift"
+      "AgentHost": "localhost",
+      "AgentPort": 6831,
+      "BatchExportProcessorOptions": {
+        "ScheduledDelayMilliseconds": 5000
+      }
+    }
+  }
+}
+```
+
+Code:
+
+```csharp
+var appBuilder = WebApplication.CreateBuilder(args);
+
+appBuilder.Services.Configure<JaegerExporterOptions>(
+    appBuilder.Configuration.GetSection("OpenTelemetry:Jaeger"));
+
+appBuilder.Services.AddOpenTelemetryTracing(
+    builder => builder.AddJaegerExporter());
+```
+
+The OpenTelemetry .NET SDK supports running multiple `TracerProvider`s inside
+the same application and it also supports registering multiple similar
+components such as exporters into a single `TracerProvider`. In order to allow
+users to target configuration at specific components a "name" parameter is
+typically supported on configuration extensions to control the options instance
+used for the component being registered.
+
+The below example shows how to configure two `JaegerExporter` instances inside a
+single `TracerProvider` sending to different ports.
+
+Json config file (usually appsettings.json):
+
+```json
+{
+  "OpenTelemetry": {
+    "JaegerPrimary": {
+      "AgentPort": 1818
+    },
+    "JaegerSecondary": {
+      "AgentPort": 8818
+    }
+  }
+}
+```
+
+Code:
+
+```csharp
+var appBuilder = WebApplication.CreateBuilder(args);
+
+appBuilder.Services.Configure<JaegerExporterOptions>(
+    "JaegerPrimary",
+    appBuilder.Configuration.GetSection("OpenTelemetry:JaegerPrimary"));
+
+appBuilder.Services.Configure<JaegerExporterOptions>(
+    "JaegerSecondary",
+    appBuilder.Configuration.GetSection("OpenTelemetry:JaegerSecondary"));
+
+appBuilder.Services.AddOpenTelemetryTracing(builder => builder
+    .AddJaegerExporter(name: "JaegerPrimary", configure: null)
+    .AddJaegerExporter(name: "JaegerSecondary", configure: null));
+```
+
+### How to set up configuration
```

Comment:
Should we move this above everything else in this section considering that this section is called "Configuration files and environment variables"?

Reply:
Updated